src/dftbp/dftbplus/initprogram.F90

```diff
@@ -1748,6 +1765,72 @@ subroutine initProgramVariables(this, input, env)
         call ThirdOrder_init(this%thirdOrd, thirdInp)
         this%cutOff%mCutOff = max(this%cutOff%mCutOff, this%thirdOrd%getCutOff())
       end if
+
+      ! Initialize multipole module
+      if (this%isDftbMultiExpan) then
+        @:ASSERT(this%tSccCalc)
+        this%dftbMultiExpanInp%orb => this%orb
+        this%dftbMultiExpanInp%nOrb = this%nOrb
+        this%dftbMultiExpanInp%nSpin = this%nSpin
+        this%dftbMultiExpanInp%nSpecies = this%nType
+
+        allocate(this%dftbMultiExpanInp%atomicDIntgrlScaling(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicQIntgrlScaling(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicOnsiteScaling(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicSXPxIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicPxXDxxyyIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicPxXDzzIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicPyYDxxyyIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicPzZDzzIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicSXXSIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicPxXXPxIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicPyXXPyIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicSXXDxxyyIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicSXXDzzIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicSYYDxxyyIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicSZZDzzIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicDxyXXDxyIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicDyzXXDyzIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicDxxyyXXDzzIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicDzzXXDzzIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicDxxyyYYDzzIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicDzzZZDzzIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicDxzXZDzzIntgrl(this%nType))
+        allocate(this%dftbMultiExpanInp%atomicDyzYZDxxyyIntgrl(this%nType))
+
+        this%dftbMultiExpanInp%atomicDIntgrlScaling(:) = input%ctrl%atomicDIntgrlScaling
```

Comment:
```suggestion
        this%dftbMultiExpanInp%atomicDIntgrlScaling = input%ctrl%atomicDIntgrlScaling
```
You can then remove the allocate() statement. Alternatively, move_allocate() could be used. Possibly a good idea to put all of this in a subroutine.